gpkit/nomials/math.py

```diff
@@ -447,74 +406,60 @@ def __init__(self, left, oper, right):
         elif self.oper == ">=":
             m_gt, p_lt = self.left, self.right
         else:
-            raise ValueError("operator %s is not supported by Posynomial"
-                             "Constraint." % self.oper)
+            raise ValueError("operator %s is not supported." % self.oper)
 
         self.p_lt, self.m_gt = p_lt, m_gt
         self.substitutions = dict(p_lt.values)
         self.substitutions.update(m_gt.values)
-        self.unsubbed = self._gen_unsubbed()
+        self.unsubbed = self._gen_unsubbed(p_lt, m_gt)
         self.nomials = [self.left, self.right, self.p_lt, self.m_gt]
         self.nomials.extend(self.unsubbed)
-        self._last_used_substitutions = None
+        self._last_used_substitutions = {}
 
-    def _simplify_posy_ineq(self, exps, cs, pmap):
+    def _simplify_posy_ineq(self, hmap, pmap=None, allow_tautological=True):
         "Simplify a posy <= 1 by moving constants to the right side."
-        coeff = 1.0
-        for i, exp in enumerate(exps):
-            if not exp:
-                cs = cs.tolist()
-                coeff -= cs.pop(i)
-                cs = np.array(cs)/coeff
-                exps_ = list(exps)
-                exps_.pop(i)
-                exps = tuple(exps_)
-                if pmap is not None:
-                    # move constant term's mmap to another attribute
-                    self.const_mmap = pmap.pop(i)  # pylint: disable=attribute-defined-outside-init
-                    self.const_coeff = coeff  # pylint: disable=attribute-defined-outside-init
-                break
-
-        if not exps and coeff >= 0:  # tautological monomial
-            return None, None, pmap
+        if EMPTY_EXP not in hmap:
+            return hmap
+        coeff = 1 - hmap[EMPTY_EXP]
+        if pmap is not None:  # note constant term's mmap
+            const_idx = hmap.keys().index(EMPTY_EXP)
+            self.const_mmap = self.pmap.pop(const_idx)  # pylint: disable=attribute-defined-outside-init
```

Comment:
heh. hmap is the NomialMap representing the nomial. it needs a new name. pmap is the post-substitution map between monomials, and const_mmap is the list of original monomials that, when substituted, became constants and were added together.